app/models/manageiq/providers/base_manager/metrics_capture.rb

```diff
@@ -1,8 +1,194 @@
+    targets.group_by(&:ems_cluster_id).each do |parent_id, targets_in_cluster|
+      start_range, end_range, counters_data = capture_targets(targets_in_cluster, interval_name, start_time, end_time)
+      perf_process_queue(ems, "perf_process", interval_name, start_range, end_range, counters_data, parent_id)
+    end
+  end
+
+  def perf_collect_targets(targets, interval_name, start_time, end_time)
+    query_size = Metric::Capture.concurrent_requests(interval_name)
+    targets.each_slice(query_size).each do |target_group|
+      start_range, end_range, counters_data = capture_targets(target_group, interval_name, start_time, end_time)
+      perf_process_queue(ems, "perf_process", interval_name, start_range, end_range, counters_data)
+    end
+  end
+
+  # Perform capture on list of targets
+  # @param targets [Array<Object>] array of targets to be captured
+  # @param start_time [String, nil] Start time
+  # @param end_time   [String, nil] Ending time
+  # [{
+  #   :ems_id         => ems_id,         # "2"
+  #   :ems_ref        => target.ems_ref, # "vm-55"
+  #   :ems_klass      => ("Vm", "Host", "Storage")
+  #   :interval_name  => ("realtime", "hourly")
+  #   :start_range    => start_time,
+  #   :end_range      => end_time,
+  #   :counters       => {
+  #     "#{counter_id}_#{instance}" => { # e.g.: "143_" =>
+  #         :counter_key           => :"#{group}_#{name}_#{stats}_#{rollup}", # :net_usage_rate_average
+  #         :rollup                => ("realtime", ...)
+  #         :precision             => (0.1, 1),
+  #         :unit_key              => ("percent", "kiloBytesPerSecond"),
+  #         :vim_key               => counter_id, # "143"
+  #         :instance              => instance,   # ""
+  #         :capture_interval      => interval,   # "20"
+  #         :capture_interval_name => ("realtime", "hourly"),
+  #       }
+  #   },
+  #   :counter_values => {
+  #     timestamp => {                           # "2017-09-25T19:55:20Z"
+  #       "#{counter_id}_#{instance}" => Numeric # "143_" => 0
+  #     }
+  #   }
+  # }]
+  # @return [Array<Hash{Symbol=>Integer,String,Array<String>,Hash{String=>Float}}>
+
+  def capture_targets(targets, interval_name, start_time, end_time)
+    all_start_range  = nil
+    all_end_range    = nil
+    all_counter_data = {}
+    targets.each do |target|
+      begin
+        capture_start, capture_end = fix_capture_start_end_time(interval_name, target, start_time, end_time)
+        start_range, end_range, counter_data = perf_capture_one(target, interval_name, capture_start, capture_end)
+        if counter_data
+          all_counter_data.merge!(counter_data)
+          all_start_range = start_range if all_start_range.nil? || all_start_range > start_range
+          all_end_range = end_range if all_end_range.nil? || all_end_range < end_range
+        end
+      rescue Handsoap::Fault => e
+        puts "problem: #{e.message}"
```

Comment:
Debugging?

Reply:
ugh.

a) what is the proper exception
b) think the other code just eats a full fledged `Exception`. Do we want to do anything special here?